common/src/chain/transaction/input.rs

```diff
@@ -23,6 +27,19 @@ impl OutPoint {
     pub fn get_output_index(&self) -> u32 {
         self.index
     }
+
+    pub fn is_coinbase(&self) -> bool {
+        self.id.is_null() && self.index == Self::COINBASE_OUTPOINT_INDEX
+    }
+}
+
+impl Idable<OutPoint> for OutPoint {
```

Comment:
Well... an OutPoint is not "ideable". If you're looking for a unique thing, use the OutPoint object itself, because it's unique, in a way.

Reply:
I wanted something that I can store in a HashSet or as a key in a HashMap. This is why I implemented this. I can't use the OutPoint itself as you say unless I implement either Hash or Ord for it. We have said Hash is out of the question. Left a question about Ord in my other comment.